### DISCUSSION TITLE
Is Next.js API route are supporting reading files?

### DISCUSSION THREAD

**Discussion by re-taro** [vercel/next.js#36914] [Help] [answered]
I'm deploying a project to Vercel and I'm using API route in the project.
I want to read files in my project's public direction but it can't works.

This is the repository -> https://github.com/re-taro/ogp.re-taro.dev

How can I works?

## Screen shots
![image](https://user-images.githubusercontent.com/68799469/168429162-b1f3cd20-f5ea-4edb-8bd7-99a7ca81cf43.png)

**Comment by GabenGar** [comment]
There is no `public` folder in the linked repo though. Where and how do you read the files?

**Comment by icyJoseph** [comment, marked as answer]
You need to use `process.cwd()` to point at the directory where the server is running. Api routes are ran as tasks, so they know nothing about your application, other than the code needed to run.

```ts
import type { NextApiRequest, NextApiResponse } from "next";
import fs from "fs/promises";
import path from "path";

type Data = {
  name: string;
};

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse<Data>
) {
  const svg = await fs.readFile(
    path.resolve(process.cwd(), "public", "vercel.svg"),
    "utf-8"
  );
  console.log(svg);
  res.status(200).json({ name: "John Doe" });
}
```

That being said, I think there was some issues including certain files in API routes. I'll search for the info and post it as reply to this answer.

Regardless, please give it a try!

**Reply by icyJoseph** [reply]
- https://vercel.com/support/articles/how-can-i-use-files-in-serverless-functions
- https://github.com/vercel/next.js/discussions/32236
